core/src/test/java/io/methvin/watchservice/DirectoryWatcherOnDiskTest.java

```diff
@@ -478,6 +476,62 @@ public void pathsWithContexts() throws IOException, InterruptedException {
     }
   }
 
+  @Test
+  public void observeIsDirectory() throws IOException, InterruptedException {
+    Path d1 = this.tmpDir.resolve("d1");
+    Files.createDirectory(d1);
+
+    this.watcher =
+          DirectoryWatcher.builder()
+                          .paths(Arrays.asList(new Path[] {d1}))
+                          .listener(this.recorder)
+                          .fileHashing(true)
```

Comment:
Is this only intended to work with hashing enabled? I'm actually not sure this will work with hashing disabled, because it depends on the file hash.